File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.cs
using System;

public class Solution 
{
    public int[] solution(int[,] score) 
    {
        int row = score.GetLength(0);
        double[] avg = new double[row];
        for(int r=0; r<row; r++) {
            avg[r] = (double)(score[r,0] + score[r,1]) / 2;    
        }
        
        int[] rank = new int[row];
        
        for(int i=0; i<rank.Length; i++) {
           rank[i] = 1;
        }
        
        for(int i=0; i<avg.Length; i++) {
            for(int j=0; j<avg.Length; j++) {
                if(avg[i] < avg[j]) {
                    rank[i]++;
                }              
            }
        }
        return rank;

    }
}

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<vector<int>> score)
{
    vector<double> avg;
    for(auto &s : score)
    {
        avg.push_back((double)(s[0] + s[1])/ 2);
    }
    
    vector<int> rank(avg.size(),1);
    for(int i=0; i<avg.size(); i++)
    {
        for(int j=0; j<avg.size(); j++)
        {
            if(avg[i] < avg[j])
            {
                rank[i]++;
            }
        }
    }
    return rank;
}